blog/_src/posts/2017-11-16-monotonicity-types-towards-a-type-system-for-eventual-consistency.md

```diff
@@ -25,37 +25,162 @@ monotonicity has arisen time and time again as an important property. Designers
 
 Thus if a user would like to make use of such a language for concurrent and distributed programming, they're required to write monotonic program functions, which can actually be quite tricky, in order to get the consistency or determinism guarantees that the given language/abstraction was designed to provide.
 
-To get a better idea of how monotonicity might be important in the context of data replicated over a distributed system, let's look at an example. Let's start with the assumption that we need a simple function that can determine whether a (potentially replicated) counter's current value is odd or even. We might write the following function to accomplish this:
+To get a better idea of how monotonicity might be important in the context of data replicated over a distributed system, let's look at an example. Suppose we need a function to determine whether a replicated counter's current value is odd or even, and further suppose that this counter can only be incremented. To accomplish this, we might apply the following function to the counter's value:
 
 ```
 fun IsOdd(x : Nat) = x % 2 == 1
 ```
 
-However, if the counter is shared among multiple hosts in a network,
-this observation could be invalid due to increments at other hosts which have not synchronized with the local one. The problem is that the value observed is not monotone with respect to the value of the counter, and so by incrementing the counter, a host can cause the value returned by IsOdd to change from true to false, without that change being visible locally. If we observe that a monotone boolean condition inferred from a replicated variable is true, we know that all hosts will eventually agree that the condition is true and, furthermore, will lack the ability to change it. It therefore can be acted upon as a global property of the system rather than a local artifact of data replication. A type system for differentiating between monotone and non-monotone conditions would therefore aid the programmer in deciding which observations they can and cannot perform on replicated data.
+However, the counter replica from which the argument x is obtained may not currently have an up-to-date count of the total number of increments performed in the entire system. We can't rule out the possibility that exactly one remote increment has been performed, in which case IsOdd produces the wrong answer. With this in mind, the value returned by IsOdd does not seem to tell us anything useful. In contrast, consider an application of the following function to the same replicated counter.
```

Comment:
I think this is much worse than what was here initially...

Reply:
You're making things less explicit, which in turns make them less intuitive and less generally-accessible to practitioners. Remember, this blog article wasn't meant for other PL PhD students or academics, it was meant for systems folks and people in industry.

Reply:
I would keep what you originally wrote, not the new text in green.